src/Common/parseGlobs.cpp

```diff
@@ -51,7 +51,15 @@ std::string makeRegexpPatternFromGlobs(const std::string & initial_str_with_glob
             ReadBufferFromString buf_range(buffer);
             buf_range >> range_begin >> point >> point >> range_end;
             bool leading_zeros = buffer[0] == '0';
-            size_t num_len = std::to_string(range_end).size();
+
+            size_t range_begin_width = buffer.find('.');
+            size_t range_end_width = buffer.size() - buffer.find_last_of('.') - 1;
+
+            size_t num_len = 0;
+            if (range_begin_width == 1 && leading_zeros)
+                num_len = 1;       /// Scenarios {0..10} {0..999}
+            else
+                num_len = range_begin_width < range_end_width ? range_end_width : range_begin_width;       /// Scenarios {00..99} {00..099}
```

Comment:
`num_len = std::max(range_begin_width, range_end_width)`

Reply:
Suggest to remove the comment here (i.e. only retain the comment in l. 60). At it confused me more than helps.